app/api/client_resources.py

```diff
@@ -115,25 +115,27 @@ def is_matching_spec(self, crd: ResourceSpec) -> bool:
             and self.alias == crd.alias
             and self.is_visible == crd.is_visible
             and self.is_browser_shortcut_enabled == crd.is_browser_shortcut_enabled
-            and self.protocols == crd.protocols
+            and self.protocols.dict() == crd.protocols.dict()
             and self.remote_network.id == crd.remote_network_id
             and (self.security_policy and self.security_policy.id)
             == crd.security_policy_id
         )
 
     def to_spec(self, **overrides: Any) -> ResourceSpec:
-        data = dict(
-            id=self.id,
-            name=self.name,
-            address=self.address.value,
-            alias=self.alias,
-            is_visible=self.is_visible,
-            is_browser_shortcut_enabled=self.is_browser_shortcut_enabled,
-            remote_network_id=self.remote_network.id,
-            security_policy_id=(
-                self.security_policy.id if self.security_policy else None
-            ),
-            protocols=self.protocols,
```

Comment:
Bug was here - `self.protocols` is an object so we need to serialize it and not send it as is.
Doing `self.protocols.dict()` here would work but a safer approach (what happens if we have another such property in the future?) is to use `self.dict(` and give it the fields we want and it'll take care of it